packages/frontend/src/components/ElectionForm/Candidates/AddCandidate.tsx

```diff
@@ -301,13 +302,77 @@ interface CandidateFormProps {
     onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void,
     electionState: string
 }
+
+const LinkDialog = ({ onEditCandidate, candidate, open, handleClose }) => {
+    const onApplyEditCandidate = (updateFunc) => {
+        const newCandidate = { ...candidate }
+        updateFunc(newCandidate)
+        onEditCandidate(newCandidate)
+    }
+
+    const [linkInput, setLinkInput] = useState(candidate.candidate_url);
+
+    return (
+        <Dialog
+            open={open}
+            onClose={handleClose}
+            scroll={'paper'}
+            keepMounted>
+            <DialogTitle> Update Hyperlink </DialogTitle>
+            <DialogContent>
+                <Box sx={{width: 300, height: 70}}>
+                    <TextField
+                        id="candidate url"
+                        label="Candidate URL"
+                        type="url"
+                        fullWidth
+                        value={linkInput}
+                        sx={{
+                            m: 1,
+                            p: 0,
+                            boxShadow: 2,
+                        }}
+                        onChange={(e) => setLinkInput(e.target.value)}
+                    />
+                </Box>
+            </DialogContent>
+
+            <DialogActions>
+                <SecondaryButton
+                    type='button'
+                    onClick={() => {
+                        onApplyEditCandidate((candidate) => { candidate.candidate_url = '' })
+                        handleClose()
+                    }}
+                >
+                    Remove
+                </SecondaryButton>
+                <PrimaryButton
+                    type='button'
+                    onClick={() => {
+                        onApplyEditCandidate((candidate) => {
+                            // Adding https:// if needed so it's not treated as a relative path
+                            candidate.candidate_url = linkInput.includes('//') ? linkInput : 'https://' + linkInput
```

Comment:
You might consider using t[he in build js URL library](https://developer.mozilla.org/en-US/docs/Web/API/URL) for a more robust solution here.